### DISCUSSION TITLE
Want reliable timestamps on (non-Lisp) files

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#620] [Yearnings]
MEDLEY-FIX-DIRS fix dates on lisp files to match their internal metadata. Would be nice in future to work with .tedit and .sysout files. This is because github doesn't remember file creation dates. But it is useful metadata.

There are a couple of ways of addressing this for non-Lisp files 
* use the SHASUM data gathered from Ron's dropbox content and add it to the calculation in MEDLEY-FIX-DIRS
* update TEDIT and SYSOUT files to have timestamps 

_Originally posted by @masinter in https://github.com/Interlisp/medley/issues/135#issuecomment-761964173_

**Comment by masinter** [comment]
Medley-fix-dirs was split into medley-fix-links and medley-fix-dates. This feature request is to make medley-fix-dates to work for sysouts and tedit files. Embedded metadata is used extensively in the creative industry with XMP.

**Comment by rmkaplan** [comment]
Does git record (and provide) the time that a file was committed, even if not the date of the file itself?

Also, the author of the commit?

**Comment by masinter** [comment]
I'm sure there is a set of options to `git log -- filename` that will display the dates and email addresses of the committer for 'filename'. It's still hard to read `git log --help`. I wanted to try setting the dates in scripts/restoreversions.sh but I couldn't figure out the options to get the commit date and set the file date to it.

**Reply by nbriggs** [reply]
I think you want to execute, at the top level of the repository:
```
git ls-files | while read x; do touch -t  `git log -1 --date=format-local:%Y%m%d%H%M.%S --pretty="format:%cd" -- $x` $x; done
```

**Reply by nbriggs** [reply]
In the `--pretty="format:%cd"`, if you're not trying to get just the date, %an and %ae give you the Author name and Author e-mail (if one was provided -- it could be something like `34669489+alex-segura@users.noreply.github.com`)

**Reply by masinter** [reply]
Thanks!  With PR #624  restore-versions.sh  will set the date/time of each version to the date/time it was committed.

**Comment by masinter** [comment]
This should have remained an Issue.
